algorithm/src/lib.rs

```diff
@@ -0,0 +1,3 @@
+#![allow(dead_code)]
```

Comment:
Rather than disabling warnings for dead code, it is better to make the module public.

```rust
pub mod algo_1
```

This means that other Rust project using this crate can use `algo_1`, and thus it is considered "being used".